zebra/zebra_rib.c

```diff
@@ -949,29 +952,28 @@ static unsigned nexthop_active_check(struct route_node *rn,
 	}
 
 	/* XXX: What exactly do those checks do? Do we support
-	 * e.g. IPv4 routes with IPv6 nexthops or vice versa? */
+	 * e.g. IPv4 routes with IPv6 nexthops or vice versa?
+	 */
 	if (RIB_SYSTEM_ROUTE(re) || (family == AFI_IP && p->family != AF_INET)
 	    || (family == AFI_IP6 && p->family != AF_INET6))
 		return CHECK_FLAG(nexthop->flags, NEXTHOP_FLAG_ACTIVE);
 
+	info = srcdest_rnode_table_info(rn);
 	/* The original code didn't determine the family correctly
 	 * e.g. for NEXTHOP_TYPE_IFINDEX. Retrieve the correct afi
 	 * from the rib_table_info in those cases.
 	 * Possibly it may be better to use only the rib_table_info
 	 * in every case.
 	 */
-	if (!family) {
-		rib_table_info_t *info;
-
-		info = srcdest_rnode_table_info(rn);
+	if (!family)
 		family = info->afi;
-	}
 
 	memset(&nexthop->rmap_src.ipv6, 0, sizeof(union g_addr));
 
 	/* It'll get set if required inside */
+	zvrf = info->zvrf;
 	ret = zebra_route_map_check(family, re->type, re->instance, p, nexthop,
-				    nexthop->vrf_id, re->tag);
+				    zvrf, re->tag);
```

Comment:
This is going to break vrf route leaking and break it badly.  Please put it back.

In fact you don't need to mess with the info pointer at all, I would just get the zvrf from either the re->vrf_id or the nexthop->vrf_id.

Reply:
done

Reply:
this still breaks vrf route leaking this needs to be the nexthop's zvrf right?

Reply:
>>>> Existing code : ret = zebra_route_map_check(family, re->type, re->instance, p, nexthop,
				    nexthop->vrf_id, re->tag);
nexthop and nexthop->vrf_id is getting passed.

>>> With this change : nexthop is getting passed and in API zebra_route_map_check() i am passing  nexthop->vrf_id to nh_obj.vrf_id
nh_obj.vrf_id = nexthop->vrf_id;

So, i have not modified this code.